src/postamble_minimal.js

```diff
@@ -3,6 +3,7 @@
 
 {{{ exportRuntime() }}}
 
+#if hasExportedFunction('_main') // Only if user is exporting a C main(), we will generate a run() function that can be used to launch main.
```

Comment:
I think `HAS_MAIN` is basically this.

Reply:
That might be subtly different based on what is assert()ed. I think I'd prefer to leave this as-is, and revisit once I start looking at adding dynamic linking support to MINIMAL_RUNTIME.

Reply:
Not sure which assertions you mean?

Thinking some more, I think hasExportedFunction is probably better than HAS_MAIN - we should check it exists *and* is exported. How about changing HAS_MAIN to do what hasExportedFunction does, and use HAS_MAIN in this PR?

Reply:
I was referring to the assertion
```js
    assert(!Module['_main'], 'compiled without a main, but one is present. if you added it from JS, use Module["onRuntimeInitialized"]');
```

Also, I am unsure how to change the code with respect to `MAIN_MODULE || SIDE_MODULE || STANDALONE_WASM` checks.

Let me just switch to HAS_MAIN in MINIMAL_RUNTIME, it is probably as close to what is needed here, even if it has a subtle difference.